Service/EnrichPubliccodeService.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace OpenCatalogi\OpenCatalogiBundle\Service;
+
+use App\Entity\Entity;
+use App\Entity\ObjectEntity;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\HttpFoundation\Response;
+use App\Entity\Gateway as Source;
+use CommonGateway\CoreBundle\Service\CallService;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Loops through repositories (https://opencatalogi.nl/oc.repository.schema.json) and and enrich them and the component (https://opencatalogi.nl/oc.component.schema.json) with fetched publiccode.yml data
+ */
+class EnrichPubliccodeService
+{
+    private EntityManagerInterface $entityManager;
+    private array $configuration;
+    private array $data;
+    private SymfonyStyle $io;
+    private CallService $callService;
+
+    private Entity $descriptionEntity;
+    private Entity $componentEntity;
+    private Entity $repositoryEntity;
+    private Source $githubApi;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        CallService $callService
+    ) {
+        $this->callService = $callService;
+        $this->entityManager = $entityManager;
+
+        $this->configuration = [];
+        $this->data = [];
+    }
+
+    /**
+     * Set symfony style in order to output to the console
+     *
+     * @param SymfonyStyle $io
+     * @return self
+     */
+    public function setStyle(SymfonyStyle $io): self
+    {
+        $this->io = $io;
+
+        return $this;
+    }
+
+    /**
+     * Updates component and repository with publiccode.yml data
+     * 
+     * @param ObjectEntity $repository
+     * @param array        $publiccode
+     *
+     * @return ObjectEntity|null dataset at the end of the handler
+     */
+    public function enrichRepositoryWithPubliccode(ObjectEntity $repository, array $publiccode = []): ?ObjectEntity
+    {
+        if (!$component = $repository->getValue('component')) {
+            $component = new ObjectEntity($this->componentEntity);
+        }
+
+        if ($publiccode !== null) {
+            if (!$description = $component->getValue('description')) {
+                $description = new ObjectEntity($this->descriptionEntity);
+            }
+
+            $description->hydrate([
```

Comment:
Dit is dus niet echt het gewenste gedrage van de hydrator functie. Graag via een mapping laten lopen (indien nodig) en dan het resultaat daardan als array aan de hydrator geven